api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
@wluyima  I did what u asked. I repeated the steps locally in the application and it turns out that even after the change no new Obs is created when an obs is voided with changes and then unvoided. Same with an obs being voided without changes and then being unvoided with changes. I have recorded the screenshots of what i performed here https://drive.google.com/drive/folders/0B54Y2dWNGuH1Y0VXbVgtZG5yVFE?usp=sharing.
I repeated the with_change procedure this time changing the actual parameters like height and weight when voiding and unvoiding and the changes didn't get saved :-) please let me know whether i followed the correct procedure